tcmur_cmd_handler.c

```diff
@@ -2485,7 +2485,7 @@ static int handle_try_passthrough(struct tcmu_device *dev,
 int tcmur_dev_update_size(struct tcmu_device *dev, unsigned long new_size)
 {
 	unsigned long old_size;
-	int ret;
+	int ret = -1;
```

Comment:
For errors we want to return a -errno value, so you could just do

ret = tcmu_update_num_lbas(dev, new_size));

However, I think that function is completely wrong or I'm not sure what we are doing.

tcmu_update_num_lbas returns -errno on failure and 0 on success, so here

```
        if (tcmu_update_num_lbas(dev, new_size)) {
                ret = tcmu_set_dev_size(dev);
```

I think instead of the above we wanted to do:

```
        if (!tcmu_update_num_lbas(dev, new_size)) {
                ret = tcmu_set_dev_size(dev);
```

where we call tcmu_set_dev_size if tcmu_update_num_lbas returns 0/success.

@lxbsz, am I right?

Reply:
Yeah, right.

        ret = tcmu_update_num_lbas(dev, new_size);
        if (!ret) {
                ret = tcmu_set_dev_size(dev);
                ...
        }

It should be something like this.